python/python/lance/dataset.py

```diff
@@ -78,6 +85,44 @@
     from .progress import FragmentWriteProgress
 
 
+class MergeInsertBuilder(object):
+    def __init__(self, dataset: LanceDataset, on: Union[str, Iterable[str]]):
+        self.dataset = dataset
+        self.builder = _MergeInsertBuilder(dataset._ds, on)
+
+    def when_matched_update_all(self) -> MergeInsertBuilder:
+        self.builder.when_matched(True)
+        return self
+
+    def when_matched_do_nothing(self) -> MergeInsertBuilder:
+        self.builder.when_matched(False)
+        return self
+
+    def when_not_matched_insert_all(self) -> MergeInsertBuilder:
+        self.builder.when_not_matched(True)
+        return self
+
+    def when_not_matched_do_nothing(self) -> MergeInsertBuilder:
+        self.builder.when_not_matched(False)
+        return self
```

Comment:
I've removed them.  As a consequence the default behavior of `merge_insert` is now to raise an error if the user doesn't specify anything further (since it would be a no-op).